cockroachdb/src/jepsen/cockroach/client.clj

```diff
@@ -224,6 +221,15 @@
         #"restart transaction"
         {:type :fail, :error [:restart-transaction m]}
 
+        ; Explicitly indicates that the operation may or may not succeed.
+        #"result is ambiguous"
+        {:type :info, :error [:result-ambiguous m]}
```

Comment:
Wouldn't it be safer to have this rule at the top of this "switch" and keep the existing `context deadline` rule? Now we're treating context deadline errors as safe even if they are not part of an ambiguous error.

Reply:
Yeah, but I figured that "context deadline exceeded" is usually an error that can happen at any layer of the stack. It is almost always ambiguous because there are many fire-and-forget steps. I.e., if the client side (or any layer of the stack) stops waiting on a response from another layer, it doesn't mean that the request/write doesn't go through and apply.

I think we should only assign "definitely not gone through" to errors that we know indicate such outcome explicitly.

Reply:
It's a good point though, we should maybe preserve the old behaviour as much as possible. If this error fires again, it would be because of a different reason, so we would investigate it additionally.

I'll send a fixup commit on top of this one.

Reply:
#38